From 50f127c1719094ba686aed2d8297c13de0e2644d Mon Sep 17 00:00:00 2001
From: Abhishek <apatgaonkar@habana.ai>
Date: Wed, 10 Jun 2020 09:55:44 +0300
Subject: [PATCH] [SW-9644] preboot: debug mode for i2c and pci module

For I2C and PCI module add debug mode which will
skip corresponding functionality in cmd module.

Change-Id: I394f198af8294f41392e09db3856ea82bf2501ee
---

diff --git a/common/module_i2c_slave.c b/common/module_i2c_slave.c
index d5c0ca6..28449e5 100644
--- a/common/module_i2c_slave.c
+++ b/common/module_i2c_slave.c
@@ -18,6 +18,7 @@
 #include "uart.h"
 #include "common.h"
 
+#define I2C_SLAVE_ADDRESS_DBG		0x47
 #define HL_DEVICE_STATUS_LINUX		0x0
 #define HL_DEVICE_STATUS_PREBOOT	0x1
 
@@ -44,14 +45,18 @@
 		return;
 	}
 
-	/* Program IC_SAR with address to which slave responds */
-	regmap_writel(I2C_SLAVE_ADDRESS, mmPSOC_I2C_S_IC_SAR);
+	if (hl_is_debug_mode(HL_DEBUG_HW_I2C)) {
+		regmap_writel(I2C_SLAVE_ADDRESS_DBG, mmPSOC_I2C_S_IC_SAR);
+	} else {
+		/* Program IC_SAR with address to which slave responds */
+		regmap_writel(I2C_SLAVE_ADDRESS, mmPSOC_I2C_S_IC_SAR);
 
-	/* Enable I2C Slave */
-	regmap_writel(0x1, mmPSOC_GLOBAL_CONF_I2C_SLV);
+		/* Enable I2C Slave */
+		regmap_writel(0x1, mmPSOC_GLOBAL_CONF_I2C_SLV);
 
-	/* Enable I2C Slave interrupts */
-	regmap_writel(0x0, mmPSOC_GLOBAL_CONF_I2C_SLV_INTR_MASK);
+		/* Enable I2C Slave interrupts */
+		regmap_writel(0x0, mmPSOC_GLOBAL_CONF_I2C_SLV_INTR_MASK);
+	}
 
 	/* Program IC_CON */
 	regmap_writel(0x22, mmPSOC_I2C_S_IC_CON);
@@ -128,6 +133,9 @@
 	enum cmd_check_res res = NOT_FOUND;
 	unsigned int stat;
 
+	if (hl_is_debug_mode(HL_DEBUG_HW_I2C))
+		return res;
+
 	i2c_dw_read_clear_intrbits_slave();
 	stat = regmap_readl(mmPSOC_I2C_S_IC_INTR_STAT);
 
diff --git a/common/module_pci.c b/common/module_pci.c
index 688bbc2..dda1a08 100644
--- a/common/module_pci.c
+++ b/common/module_pci.c
@@ -26,7 +26,8 @@
 
 void module_pci_setup(void)
 {
-	pci_dev_specific_setup();
+	if (!hl_is_debug_mode(HL_DEBUG_HW_PCIE))
+		pci_dev_specific_setup();
 }
 
 __attribute__((weak)) enum cmd_check_res dev_specific_module_pci_check(void)
@@ -42,6 +43,9 @@
 
 enum cmd_check_res module_pci_check(void)
 {
+	if (hl_is_debug_mode(HL_DEBUG_HW_PCIE))
+		return NOT_FOUND;
+
 	return dev_specific_module_pci_check();
 }
 
